src/bsoncxx/oid.hpp

```diff
@@ -37,27 +37,10 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
 ///
 class BSONCXX_API oid {
    public:
-    struct init_tag_t {};
-
-    // TODO(MSVC): Ideally this would be constexpr, but VS2015U1 can't
-    // handle it.
-    //
-    // See https://connect.microsoft.com/VisualStudio/feedback/details/2092790
-    //
-    static const init_tag_t init_tag;
-
-    ///
-    /// Constructs an uninitialized oid.
-    ///
-    oid();
-
     ///
     /// Constructs an oid and initializes it to a newly generated ObjectId.
     ///
-    /// @param tag
-    ///   A bsoncxx::oid::init_tag used to dispatch this overload.
-    ///
-    explicit oid(init_tag_t tag);
+    oid();
 
     ///
     /// Constructs an oid initializes it to the contents of the provided buffer.
```

Comment:
Use the source Luke =)

https://github.com/ajdavis/libbson/blob/c7c8109f87e3626e7dc5ce4281ceeda48ebf9762/src/bson/bson-oid.h#L214

`bson_oid_init_from_string` doesn't care whether you null terminate its input